mlflow/utils/file_utils.py

```
@@ -676,42 +684,46 @@ def parallelized_download_file_using_http_uri(
 
     def run_download(range_start, range_end):
         with tempfile.TemporaryDirectory() as tmpdir:
-            temp_file = os.path.join(tmpdir, "error_messages.txt")
-            download_proc = _exec_cmd(
-                cmd=[
-                    sys.executable,
-                    download_cloud_file_chunk.__file__,
-                    "--range-start",
-                    range_start,
-                    "--range-end",
-                    range_end,
-                    "--headers",
-                    json.dumps(headers or {}),
-                    "--download-path",
-                    download_path,
-                    "--http-uri",
-                    http_uri,
-                    "--temp-file",
-                    temp_file,
-                ],
-                throw_on_error=True,
-                synchronous=False,
-                capture_output=True,
-                stream_output=False,
-                env=env,
-            )
-            _, stderr = download_proc.communicate(timeout=MLFLOW_DOWNLOAD_CHUNK_TIMEOUT.get())
-            if download_proc.returncode != 0:
-                if os.path.exists(temp_file):
-                    with open(temp_file) as f:
-                        file_contents = f.read()
-                        if file_contents:
-                            return json.loads(file_contents)
-                        else:
-                            raise Exception(
-                                "Error from download_cloud_file_chunk not captured, "
-                                f"return code {download_proc.returncode}, stderr {stderr}"
-                            )
```

The original code swallows non-HTTP errors.